cf-agent/files_editline.c

```diff
@@ -1656,11 +1657,26 @@ static bool MatchPolicy(EvalContext *ctx, const char *camel, const char *haystac
                     char  *work      = xcalloc(1, work_size);
                     strcpy(work, sp);
 
-                    if (snprintf(final, final_size, "\\s*%s", work) >= final_size - 1)
+                    int written = snprintf(final, final_size, "\\s*%s", 
+                                           work);
+                    if (written < 0)
+                    {
+                        Log(LOG_LEVEL_ERR, "snprintf failed: %s", 
```

Comment:
Imagine debugging a failure like this. All you'll get is `snprintf failed: SOME REASON` and there will be no clue where and what the `snprintf()` was even trying to format. We need to add the information like that, preferably with some wording understandable to the users if they are likely to see these errors.